RFCs/FS-1063-support-letbang-andbang-for-applicative-functors.md

```diff
@@ -0,0 +1,432 @@
+
+In order to keep things simple, then, we keep the canonical form introduced earlier, which forces precisely one `return` after a `let! ... and! ...`. However, we can still express alternative applicatives when using the `let! ... and! ...` syntax, we just need to use the same trick as with additional `let!`s and leave the scope of the canonical applicative syntax and therefore leave the additional constraints it places upon us:
+
+```fsharp
+ce {
+    yield
+        ce {
+            let! x = foo
+            and! y = bar
+            and! z = baz
+            return x + y + z
+        }
+    yield
+        ce {
+            let! x = foo
+            and! y = bar
+            return x + y
+        }
+    yield
+        ce {
+            let! x = foo
+            and! y = bar
+            return y + z
+        }
+}
+```
+
+?
+
+```fsharp
+ce.Combine(
+    ce.Combine(
+        ce.Apply(
+            ce.Apply(
+                ce.Apply(
+                    ce.Return(
+                        (fun x ->
+                            (fun y ->
+                                (fun z ->
+                                    x + y + z
+                                )
+                            )
+                        )),
+                    foo),
+                bar),
+            baz),
+        ce.Apply(
+            ce.Apply(
+                ce.Return(
+                    (fun x ->
+                        (fun y ->
+                            x + y
+                        )
+                    )),
+                foo),
+            bar)
+    ),
+    ce.Apply(
+        ce.Apply(
+            ce.Return(
+                (fun y ->
+                    (fun z ->
+                        y + z
+                    )
+                )),
+            bar),
+        baz)
+)
+```
+
+*N.B.* this syntax forces the writer to be explicit about how many times `Apply` should be called, and with which arguments, for each call to `Combine`. Notice also how the right-hand sides are still repeated for each alternative case in order to keep the occurrence of potential side-effects from evaluating them predictable, and also occur before the pattern matching _each time_ a new alternative case is explored.
+
+If this syntax feels a bit heavy, remember that the `yield` keyword is not required, and the new syntax can be mixed with other styles (e.g. a custom `<|>` alternation operator) to strike an appropriate balance as each situation requires.
+
+## Using
+
+Just as monads support `Using` via `use!`, applicatives supports it via `use! ... anduse! ...`. Each binding can be either `and!` or `anduse!` (unless it is the first, in which case it must be either `let!` or `use!`), i.e. you can mix-and-match to describe which bindings should be covered by a call to `Using`:
```

Comment:
`anduse!` is interesting, since it does sort of break the visual symmetry you can get with `let!`, `use!`, and `and!`. I'm not opposed to it, mostly because I don't have a good alternative, but it does feel a bit odd. Limitation of the English language, I suppose.

Reply:
I totally agree. I've asked for alternative suggestions for this keyword at the bottom of the RFC and the discussion. It would be nice to have something a bit more tidy here.